tests/foreman/virtwho/api/test_esx_sca.py

```diff
@@ -0,0 +1,500 @@
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        options = {'true': '1', 'false': '0', '1': '1', '0': '0'}
+        for key, value in sorted(options.items(), key=lambda item: item[0]):
+            virtwho_config.debug = key
+            virtwho_config.update(['debug'])
+            command = get_configure_command(virtwho_config.id, module_sca_manifest_org.name)
+            deploy_configure_by_command(
+                command, form_data['hypervisor_type'], org=module_sca_manifest_org.label
+            )
+            assert get_configure_option('debug', ETC_VIRTWHO_CONFIG) == value
+
+    @pytest.mark.tier2
+    def test_positive_interval_option(
+        self, module_sca_manifest_org, form_data, virtwho_config, target_sat
+    ):
+        """Verify interval option by "PUT
+
+        /foreman_virt_who_configure/api/v2/configs/:id"
+
+        :id: aa9c7f19-c402-4197-9d3a-1379d9126620
+
+        :expectedresults: interval option can be updated.
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        options = {
+            '60': '3600',
+            '120': '7200',
+            '240': '14400',
+            '480': '28800',
+            '720': '43200',
+            '1440': '86400',
+            '2880': '172800',
+            '4320': '259200',
+        }
+        for key, value in sorted(options.items(), key=lambda item: int(item[0])):
+            virtwho_config.interval = key
+            virtwho_config.update(['interval'])
+            command = get_configure_command(virtwho_config.id, module_sca_manifest_org.name)
+            deploy_configure_by_command(
+                command, form_data['hypervisor_type'], org=module_sca_manifest_org.label
+            )
+            assert get_configure_option('interval', ETC_VIRTWHO_CONFIG) == value
+
+    @pytest.mark.tier2
+    def test_positive_hypervisor_id_option(
+        self, module_sca_manifest_org, form_data, virtwho_config, target_sat
+    ):
+        """Verify hypervisor_id option by "PUT
+
+        /foreman_virt_who_configure/api/v2/configs/:id"
+
+        :id: fb1ebabe-7522-45a3-8e27-05101c50aabe
+
+        :expectedresults: hypervisor_id option can be updated.
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        values = ['uuid', 'hostname']
+        for value in values:
+            virtwho_config.hypervisor_id = value
+            virtwho_config.update(['hypervisor_id'])
+            config_file = get_configure_file(virtwho_config.id)
+            command = get_configure_command(virtwho_config.id, module_sca_manifest_org.name)
+            deploy_configure_by_command(
+                command, form_data['hypervisor_type'], org=module_sca_manifest_org.label
+            )
+            assert get_configure_option('hypervisor_id', config_file) == value
+
+    @pytest.mark.tier2
+    def test_positive_filter_option(self, module_sca_manifest_org, form_data, target_sat):
```

Comment:
this is one impressive test! do you think it would be possible to break it up with parametrization? I see a lot of repeated logic which typically indicates a good candidate for splitting/parametrizing.

Reply:
> this is one impressive test! do you think it would be possible to break it up with parametrization? I see a lot of repeated logic which typically indicates a good candidate for splitting/parametrizing.

@JacobCallahan Have already parametrized tests.
```
(robottelo_vv) [root@dell-per740-68-vm-04 robottelo_20230307]# pytest  ./tests/foreman/virtwho/api/test_esx_sca.py  -k test_positive_filter_option  --disable-pytest-warnings -q
....                                                                                                                                                                                                        [100%]
4 passed, 12 deselected, 23 warnings in 234.93s (0:03:54)
```